packages/fury-adapter-remote/lib/adapter.js

```diff
@@ -0,0 +1,131 @@
+const deckardcain = require('deckardcain');
+const axios = require('axios');
+
+const name = 'remote';
+
+const outputMediaType = 'application/vnd.refract.parse-result+json; version=1.0';
+
+const defaultOptions = {
+
+  // the default value, but consumers should be able to override to use their own deployment
+  url: 'https://api.apielements.org',
+
+  parseEndpoint: '/parser',
+  validateEndpoint: '/validate',
+  serializeEndpoint: '/composer',
+
+  // the collection of "parse", media types we want this
+  // instance of the adapter to handle.
+  // NOTE, this allows you to use the API for one media type but
+  // another local adapter for another.
+
+  mediaTypes: [
```

Comment:
The design in https://github.com/apiaryio/api-elements.js/issues/143#issue-406774120 separates the media types for parsing and serialisation, the reason for this is that you might want to configure the adapter to only parse API Blueprint but not support serializing API Blueprint so the media types would need to be separate arrays:

```
  // the collection of "parse", media types we want this
  // instance of the adapter to handle.

  // NOTE, this allows you to use the API for one media type but
  // another local adapter for another.
  parseMediaTypes: ['text/vnd.apiblueprint', 'application/oai.oas'],

  // the collection of "serialize", media types we want this
  // instance of the adapter to handle.
  serializeMediaTypes: ['text/vnd.apiblueprint'],
```

Reply:
I agree, but it is what i wrote in initial comment.
`detect()` is not able to check kind of operation you are asking for mediaType and same operation itself. If I remove it, it not will be able to serialisation later.

Maybe I can split it in for public space (to avoid changing interface in future), but internaly in `detect` I must do check for both.

```
return this.parseMediaTypes.includes(deckardcain.identify(source)) || serializeMediaTypes.includes(deckardcain.identify(source));
```

Seems to good solution for you?